typescript/listeners/format.py

```diff
@@ -24,8 +24,10 @@ def on_modified_with_info(self, view, info):
             info.prev_sel[0].empty()
         ):
             last_command, args, repeat_times = view.command_history(0)
+            redo_command = view.command_history(1)[0]
             log.debug("last_command:{0}, args:{1}".format(last_command, args))
-            if last_command == "insert":
+            log.debug("redo_command:{0}".format(redo_command))
+            if last_command == "insert" and redo_command != "typescript_format_on_key":
```

Comment:
We may want to rethink this approach - if there are any items to redo, then that means that the user just performed an undo, right? If there's anything that can be redone, then the user is traversing between undos and redos, and we should just hold back on doing anything.

Though I'm realizing, if a user redoes all of their operations, there is nothing on the redo list - but we wouldn't want to immediately format. Let's discuss this a bit.

__________________

If we do decide to go with the existing code,  I'd prefer you defined `redo_command` and performed the check inside this if, and put a comment before it like

```Python
# In this case, the user has requested we undo a formatting operation.
# In this case, 
redo_command = view.command_history(1)[0]
if redo_command == "typescript_format_on_key":
    return
```